### PR TITLE
Fixed PR 946 issues

### PR DESCRIPTION
**Pull request by orenbm**
#### What does this PR do?
fixes comments from PR #946 

- Added log for certificate fetch
- changed `certs` method to `fetch-certs` to be action-oriented

### CODE REVIEW

File: app/domain/authentication/authn_oidc/authenticate_id_token/decode_and_verify_id_token.rb [hunk 1]
```diff
@@ -42,7 +42,7 @@ def verify_decoded_id_token
           raise IdTokenVerifyFailed, e.inspect
         end
 
-        def certs
+        def fetch_certs
           @certs ||= @provider_certificate.fetch_certs(@provider_uri)
```
Comment (jtuttle) [inline app/domain/authentication/authn_oidc/authenticate_id_token/decode_and_verify_id_token.rb]: This doesn't actually need to be memoized. Just set it using `@certs =` and then refer to it as `@certs` in the method below. That should hopefully remove the CC warnings.

**Comment by jtuttle** [comment]
I just spoke w/ Jonah a bit about the CodeClimate errors and he suggested turning `ProviderCertificate` into a `FetchCertificates` command class. This would allow you to inject `Rails.logger` as a dependency which would solve one of the errors. Mind doing that and then we'll deal with the other one?

File: app/domain/authentication/authn_oidc/authenticate_id_token/fetch_provider_certificate.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        end
+
+        private
+
+        def discover_provider
+          @logger.debug("[OIDC] Discovering provider '#{@provider_uri}'")
+
+          @discovered_provider = OpenIDConnect::Discovery::Provider::Config.discover!(@provider_uri)
+        rescue HTTPClient::ConnectTimeoutError => e
+          raise ProviderDiscoveryTimeout.new(@provider_uri, e.inspect)
```
Comment (jtuttle) [inline app/domain/authentication/authn_oidc/authenticate_id_token/fetch_provider_certificate.rb]: You could silence this `reek` error by creating a `raise_error` method:

```
def raise_error(error_class, original_error)
  raiser error_class.new(@provider_uri, original_error.inspect)
end
```

Comment (orenbm) [reply app/domain/authentication/authn_oidc/authenticate_id_token/fetch_provider_certificate.rb]: done

**Pull request merged**